vortex-array/src/patches.rs

```diff
@@ -258,7 +300,99 @@ impl Patches {
             PrimitiveArray::from_vec(values_indices, Validity::NonNullable).into_array();
         let new_values = take(self.values(), values_indices)?;
 
-        Ok(Some(Self::new(indices.len(), new_indices, new_values)))
+        Ok(Some(Self::new(new_length, new_indices, new_values)))
+    }
+
+    pub fn take_map(&self, take_indices: &ArrayData) -> VortexResult<Option<Self>> {
```

Comment:
This shouldn't be a public function imho, having users decide how they want take to be peformed seems like a footgun. We already have a threshold for two implementations, just make this private

Reply:
Agreed, but see above. I need it accessible to the benchmarks. I could feature flag some public shims around private methods. What do you think of that?